source/adapters/level_zero/adapter.cpp

```diff
@@ -144,6 +144,43 @@ ur_result_t adapterStateInit() {
   return UR_RESULT_SUCCESS;
 }
 
+/*
+This constructor initializes the `ur_adapter_handle_t_` object and
+sets up the environment for Level Zero (L0) initialization.
+The behavior of the initialization process is influenced by two
+environment variables:
+`UR_L0_ENABLE_SYSMAN_ENV_DEFAULT` and `UR_L0_ENABLE_ZESINIT_DEFAULT`.
+
+| Environment Variable           | Value | Behavior                   |
+|--------------------------------|-------|----------------------------|
+| UR_L0_ENABLE_SYSMAN_ENV_DEFAULT| 1     | Enables the default SysMan |
+|                                |       | environment initialization |
+|                                |       | by setting                 |
+|                                |       | `ZES_ENABLE_SYSMAN` to "1".|
+|                                | 0     | Disables the default SysMan|
+|                                |       | environment initialization.|
+|                                | unset | Defaults to 1, enabling the|
+|                                |       | SysMan environment         |
+|                                |       | initialization.            |
+| UR_L0_ENABLE_ZESINIT_DEFAULT   | 1     | Enables the default SysMan |
+|                                |       | initialization by loading  |
+|                                |       | SysMan-related functions   |
+|                                |       | and calling `zesInit`.     |
+|                                | 0     | Disables the default SysMan|
+|                                |       | initialization with zesInit|
+|                                | unset | Defaults to 0, disabling   |
+|                                |       | the SysMan initialization  |
+|                                |       | thru zesInit.              |
```

Comment:
I feel like these should be described in the [env var docs](https://oneapi-src.github.io/unified-runtime/core/INTRO.html#environment-variables) but I don't see any of the other L0 adapter env vars their either.

I think the best place for these would be an L0 reference doc, similar to the [CUDA ref doc](https://oneapi-src.github.io/unified-runtime/core/CUDA.html) and [HIP ref doc](https://oneapi-src.github.io/unified-runtime/core/HIP.html) describing all the L0 adapter specific env vars. This could be done in a separate PR, though.